src/halmos/__main__.py

```diff
@@ -885,18 +885,33 @@ def _solve_end_to_end_callback(
                 return
 
             error(f"encountered exception during assertion solving: {e!r}")
```

Comment:
this error can also be triggered:

ERROR    encountered exception during assertion solving: OSError(9, 'Bad file descriptor')